python/ray/tune/suggest/search.py

```diff
@@ -24,7 +24,7 @@ def metric(self):
         return self._metric
 
     def set_search_properties(self, metric: Optional[str], mode: Optional[str],
-                              config: Dict) -> bool:
+                              config: Dict, **spec) -> bool:
```

Comment:
IMO the base class should include all default arguments we actually pass to the child classes.

The child classes can then have `**spec` for forward compatibility

Reply:
What's the downside of having **spec in base signature as well? If one is to implement a customized SearchAlgorithm, one should be able to figure out what is the right interface to implement against by looking at base SearchArgorithm class.

Reply:
Yes, that's why I would suggest to put `experiment` here, too (so people see that it is passed to searchers). We should also have `**spec` in that signature so when people inherit from this class they will keep the signature and hence keep forward compatibility

Reply:
oh I see! Sorry was confused.
See Antoni's comment above (and Richard). I think we settled down on not passing experiment around and just use kwargs.